src/components/Modal.tsx

```diff
@@ -69,6 +69,14 @@ const ModalPanel = styled(Box)`
 	box-shadow: 0px 0px 15px 1px rgba(0, 0, 0, 0.4);
 `;
 
+const ModalButton = (props: ButtonProps | ButtonAnchorProps) => {
+	return 'href' in props && props.href ? (
```

Comment:
Why do we need the `'href' in props` part? Seems redundant to me, but I might be missing something.

Reply:
@sradevski after `'href' in props` TS can infer that `props` are of `ButtonAnchorProps`. W/o it we would need to cast the props before the check `(props as ButtonAnchorProps).href`, otherwise we would get a `'href' does not exist on type 'ButtonProps'` error.